### PR TITLE
 seafile-client: 6.1.8 -> 6.2.1

### LINKED ISSUE #42772: seafile-client 6.1.8 fails to build
## Issue description

seafile-client 6.1.8 fails to build at cmake configuration step with the following error message:

```
CMake Error at CMakeLists.txt:802 (QT5_USE_MODULES):
  Unknown CMake command "QT5_USE_MODULES".
```

### Steps to reproduce

run `nix-env -iA nixos.seafile-client`

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.51, NixOS, 18.09pre144572.9720833743f (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos-18.09pre143771.a8c71037e04, nixpkgs-18.09pre144572.9720833743f, unstable-small-18.09pre144572.9720833743f"`
 - channels(nek0): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
fixes #42772 

~Note that this is untested due to the issue with running applications built against different Qt versions.~

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by xeji** [comment]
Source hash mismatch: 

```
fixed-output derivation produced path '/nix/store/a8c958yhzma20xk6gkfg0lvba84jli88-source' with sha256 hash '199ivh6fbz4sgbz15mr6m8985yky5n3z6vpcpnh2ldjx4277gq68' instead of the expected hash '1qnsk0s1j74zx1h7z1cclamknfa74gbspj12g9lbi7all2s14943'
```

**Comment by dotlambda** [comment]
Now I also tested the package and can confirm it works as expected :tada:

File: pkgs/development/libraries/libsearpc/default.nix [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-  src = fetchurl
-  {
-    url = "https://github.com/haiwen/libsearpc/archive/v${version}.tar.gz";
-    sha256 = "0fdrgksdwd4qxp7qvh75y39dy52h2f5wfjbqr00h3rwkbx4npvpg";
+  src = fetchFromGitHub {
+    owner = "haiwen";
+    repo = "libsearpc";
+    # Tag is missing: https://github.com/haiwen/libsearpc/commit/12a01268825e9c7e17794c58c367e3b4db912ad9
+    rev = "12a01268825e9c7e17794c58c367e3b4db912ad9";
+    sha256 = "00ck1hl1x0pn22q3ba32dq3ckc4nrsg58irsmrnmalqbsffhcim0";
```
Comment (orivej) [inline pkgs/development/libraries/libsearpc/default.nix]: It seems that with their rolling `v3.1-latest` tag upstream wants us to ignore releases and always use the latest commit.

Comment (orivej) [reply pkgs/development/libraries/libsearpc/default.nix]: Although the difference between this commit and the latest is not important; feel free to keep using it.

Comment (dotlambda) [reply pkgs/development/libraries/libsearpc/default.nix]: I just did what Arch (AUR) does: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=libsearpc

**Pull request merged**